### PR TITLE
Bump k8s client to k8s-1.9.x

### PR DESCRIPTION
**Pull request by imikushin**
Vendor riff dependencies separately from the rest of the project as it does not compile against the new k8s.io/client-go version.

Encapsulate riff dependencies in `lib/riff/vendor`: `lib/riff` does not depend on any dispatch or
third party packages except `github.com/pkg/errors` and `github.com/sirupsen/logrus`.

### CODE REVIEW

File: Gopkg.lock [hunk 1]
```diff
@@ -797,10 +784,12 @@
   revision = "eb3733d160e74a9c7e442f435eb3bea458e1d19f"
 
 [[projects]]
-  branch = "release-1.8"
```
Comment (berndtj) [inline Gopkg.lock]: FYI, I'll be bumping these to `version = "kubernetes-1.9.1"`

**Comment by berndtj** [comment]
I don't have a problem with this change, let's just get the tests to pass.

Also, in I've confirmed with @markpeek that we should move the faas drivers to their own repositories under github.com/dispatchframework.

File: Gopkg.lock (hunk 1 above)
Comment (imikushin) [reply Gopkg.lock]: Why not the latest stable in the 1.9.x branch?

Comment (berndtj) [reply Gopkg.lock]: Isn't release more "stable" than HEAD on a branch? I'm not sure that I care, just that the client-go package is using branch = "release-6.0" which isn't clear which k8s version it supports.  Tagging based on release just seems more straight forward and definitive.

**Comment by imikushin** [comment]
Factoring out faas drivers will be the next step

**Pull request closed without merging**